src/backend/distributed/replication/multi_logical_replication.c

```diff
@@ -1418,33 +1705,27 @@ DropShardUser(MultiConnection *connection, char *username)
 
 
 /*
- * CreateShardMovePublications creates a set of publications for moving a list
- * of shards over the given connection. One publication is created for each of
- * the table owners in tableOwnerIds. Each of those publications only contains
- * shards that the respective table owner owns.
+ * CreatePublications creates a the publications defined in the
+ * publicationInfoHash over the given connection.
  */
-static void
-CreateShardMovePublications(MultiConnection *connection, List *shardList,
-							Bitmapset *tableOwnerIds)
+void
+CreatePublications(MultiConnection *connection,
+				   HTAB *publicationInfoHash)
 {
```

Comment:
To facilitate cleanup, we would want to create the publication and metadata entry about the artifact(replication slot too) in the same transaction.
I think it would be better to add `OperationType(MOVE/SPLIT)` in all the methods that we are using in common between split and move.
For eg: CreatePublications, CreateReplicationSlots at least. 
To facilitate granular cleanup, the cleanup process can only look an artitfact for specific type of operation and work on that.This would also allow a conncurrent cleanup of split vs concurrent move operation.

Reply:
As discussed over a call, we'll do this when needed (and hopefully its not needed).